source/NVDAObjects/IAccessible/sysListView32.py

```diff
@@ -266,19 +269,27 @@ def _getColumnOrderArrayRawOutProc(self, columnCount: int) -> Optional[ctypes.Ar
 			if res:
 				winKernel.readProcessMemory(processHandle,internalCoa,byref(coa),sizeof(coa),None)
 			else:
-				log.debugWarning("LVM_GETCOLUMNORDERARRAY failed for list")
+				coa = None
+				log.debugWarning(
+					f"LVM_GETCOLUMNORDERARRAY failed for list. "
+					f"Windows Error: {ctypes.GetLastError()}, Handle: {self.windowHandle}"
+				)
 		finally:
 			winKernel.virtualFreeEx(processHandle,internalCoa,0,winKernel.MEM_RELEASE)
 		return coa
 
 	def _getColumnOrderArrayRaw(self, columnCount: int) -> Optional[ctypes.Array]:
 		"""Retrieves an array of column indexes for a given list.
 		The indexes are placed in order in which columns are displayed on screen from left to right.
-		Note that when columns are reordered the indexes remain  the same - only their ordder differs.
+		Note that when columns are reordered the indexes remain the same - only their order differs.
 		"""
-		if not self.appModule.helperLocalBindingHandle:
-			return self._getColumnOrderArrayRawOutProc(columnCount)
-		return self._getColumnOrderArrayRawInProc(columnCount)
+		if self.appModule.helperLocalBindingHandle is not None:
+			inProcessArray = self._getColumnOrderArrayRawInProc(columnCount)
+			if inProcessArray is not None:
+				return inProcessArray
+		return self._getColumnOrderArrayRawOutProc(columnCount)

```

Comment:
I'm not sure what situation would result with us being able to inject in process (I.E. `helperLocalBindingHandle` is set) unable to get a result for `ColumnOrderArray` in process yet able to get a result out of process.

I'm concerned that the most likely outcome here is for a failure (like the 'start all back' case) encounters both failures, slowing NVDA down for no result.

Perhaps instead:
```suggestion
		if self.appModule.helperLocalBindingHandle is not None:
			inProcessArray = self._getColumnOrderArrayRawInProc(columnCount)
			return inProcessArray
		return self._getColumnOrderArrayRawOutProc(columnCount)

```

Reply:
I've updated the code, PR title and description to reflect this change.